pkg/registry/query.go

```diff
@@ -30,20 +31,29 @@ func (q Querier) ListPackages(_ context.Context) ([]string, error) {
 }
 
 func (q Querier) ListBundles(_ context.Context) ([]*api.Bundle, error) {
-	var bundles []*api.Bundle
+	sliceStreamer := streamer.NewSliceStreamer()
 
+	err := q.StreamBundles(context.TODO(), sliceStreamer)
+	if err != nil {
+		return nil, err
+	}
+
+	return sliceStreamer.Slice, nil
+}
+
+func (q Querier) StreamBundles(_ context.Context, s streamer.Stream) error {
 	for _, pkg := range q.pkgs {
 		for _, ch := range pkg.Channels {
 			for _, b := range ch.Bundles {
 				apiBundle, err := api.ConvertModelBundleToAPIBundle(*b)
 				if err != nil {
-					return nil, fmt.Errorf("convert bundle %q: %v", b.Name, err)
+					return fmt.Errorf("convert bundle %q: %v", b.Name, err)
 				}
-				bundles = append(bundles, apiBundle)
+				s.Send(apiBundle)
```

Comment:
Does this return an error that should be checked?

Also, do we need to check whether the context is `Done()` before each `Send()`? 

It seems like the GRPC server has it's own context that will be plumbed into this function, so I'm not really sure if `s.Send` will handle a context that has been cancelled. It's odd to me that this API does not accept a context directly.

Reply:
+1, the error needs to be checked.

>  do we need to check whether the context is Done() before each Send()?

My (unverified) assumption is that Send returns an error (e.g. ContextCancelled) in these cases and can be treated as an implementation detail.

Reply:
Yeah I was trying to find the actual implementation of that `Send` call, but I quickly hit a dead end trying to track that down through generated code and interfaces.

I think what I'm slightly worried about is that the context that is passed in isn't necessarily tied at all to the stream's context. 

It looks like `Send()` blocks in various situations, so this is probably not really a solvable problem for us. Even if we check our context's `Done()` before every `Send()`, `Send()` can block forever and never let us find out we later have a cancelled context.

TL;DR: I think its fine to just handle the error and not worry about the context that is passed in.

Reply:
> I think what I'm slightly worried about is that the context that is passed in isn't necessarily tied at all to the stream's context.

That sounds accurate to me, but IMO that's up to the caller, not the Querier implementation. In the RegistryServer case, we know that the contexts should be linked (identical) because we're passing `stream.Context()` to ListBundles, but it definitely looks possible to use this in such a way that you could cancel the ListBundles Context and go on to hang indefinitely on a Sender implementation that does not respect that cancellation.